src/python/qeopenfermion/_utils.py

```diff
@@ -296,45 +296,61 @@ def reverse_qubit_order(qubit_operator:QubitOperator, n_qubits:Optional[int]=Non
-			Pyquil.
-	Returns:
-		complex: the expectation value
-	"""
-	n_qubits = wavefunction.amplitudes.shape[0].bit_length() - 1
-	
-	# Convert the qubit operator to a sparse matrix. Note that the qubit indices
-	# must be reversed because OpenFermion and pyquil use different conventions
-	# for how to order the computational basis states!
-	if reverse_operator:
-		qubit_op = reverse_qubit_order(qubit_op, n_qubits=n_qubits)
-	sparse_op = get_sparse_operator(qubit_op, n_qubits=n_qubits)
-	
-	# Computer the expectation value
-	exp_val = openfermion_expectation(sparse_op, wavefunction.amplitudes)
-	return exp_val
-	
+    """Get the expectation value of a qubit operator with respect to a wavefunction.
+    Args:
+        qubit_op (openfermion.ops.QubitOperator): the operator
+        wavefunction (pyquil.wavefunction.Wavefunction): the wavefunction
+        reverse_operator (boolean): whether to reverse order of qubit operator
+            before computing expectation value. This should be True if the convention
+            of the basis states used for the wavefunction is the opposite of the one in
+            the qubit operator. This is the case, e.g. when the wavefunction comes from
+            Pyquil.
+    Returns:
+        complex: the expectation value
+    """
+    n_qubits = wavefunction.amplitudes.shape[0].bit_length() - 1
+    
+    # Convert the qubit operator to a sparse matrix. Note that the qubit indices
+    # must be reversed because OpenFermion and pyquil use different conventions
+    # for how to order the computational basis states!
+    if reverse_operator:
+        qubit_op = reverse_qubit_order(qubit_op, n_qubits=n_qubits)
+    sparse_op = get_sparse_operator(qubit_op, n_qubits=n_qubits)
+    
+    # Computer the expectation value
+    exp_val = openfermion_expectation(sparse_op, wavefunction.amplitudes)
+    return exp_val
+    
 
 def change_operator_type(operator, operatorType):
-	'''Take an operator and attempt to cast it to an operator of a different type
-
-	Args:
-		operator: The operator
-		operatorType: The type of the operator that the original operator is
-			cast to
-	Returns:
-		An operator with type operatorType
-	'''
-	new_operator = operatorType()
-	for op in operator.terms:
-		new_operator += operatorType(tuple(op), operator.terms[op])
-	
-	return new_operator
+    '''Take an operator and attempt to cast it to an operator of a different type
+
+    Args:
+        operator: The operator
+        operatorType: The type of the operator that the original operator is
+            cast to
+    Returns:
+        An operator with type operatorType
+    '''
+    new_operator = operatorType()
+    for op in operator.terms:
+        new_operator += operatorType(tuple(op), operator.terms[op])
+    
+    return new_operator
+
+def get_fermion_number_operator(n_qubits, n_particles=None):
+    """Return a FermionOperator representing the number operator
```

Comment:
Maybe add to the function description what happens if n_particles is specified, and the same in the documentation. Maybe we could even say something like "specify n_particles when using the number operator as an equality constraint"